megameklab/src/megameklab/ui/MegaMekLabTabbedUI.java

```diff
@@ -0,0 +1,352 @@
+     * @param filename The name of the file associated with the unit being added.
+     */
+    public void addUnit(Entity entity, String filename) {
+        // Create a new "new tab" button, since we're about to replace the existing one
+        addNewButton();
+        // Select the old "new tab" button...
+        tabs.setSelectedIndex(tabs.getTabCount() - 2);
+        // ...and replace it, since newUnit is actually the Switch Unit Type operation.
+        newUnit(UnitUtil.getEditorTypeForEntity(entity), entity.isPrimitive(), entity.isIndustrialMek());
+
+        currentEditor().setEntity(entity, filename);
+        currentEditor().reloadTabs();
+        currentEditor().refreshAll();
+        // Set the tab name
+        tabs.setTabComponentAt(tabs.getSelectedIndex(), new ClosableTab(entity.getDisplayName(), currentEditor()));
+    }
+
+    @Override
+    public boolean exit() {
+        if (!currentEditor().safetyPrompt()) {
+            return false;
+        }
+
+        CConfig.setParam(CConfig.GUI_FULLSCREEN, Integer.toString(getExtendedState()));
+        CConfig.setParam(CConfig.GUI_PLAF, UIManager.getLookAndFeel().getClass().getName());
+        CConfig.writeMainUiWindowSettings(this);
+        CConfig.saveConfig();
+        PreferenceManager.getInstance().save();
+        MegaMek.getMMPreferences().saveToFile(MMLConstants.MM_PREFERENCES_FILE);
+        MegaMekLab.getMMLPreferences().saveToFile(MMLConstants.MML_PREFERENCES_FILE);
+
+        if (CConfig.getStartUpType() == MMLStartUp.RESTORE_TABS) {
+            try {
+                TabStateUtil.saveTabState(editors.stream().limit(editors.size() - 1).toList());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        return true;
+    }
+
+    private void restrictToScrenSize() {
+        DisplayMode currentMonitor = getGraphicsConfiguration().getDevice().getDisplayMode();
+        int scaledMonitorW = UIUtil.getScaledScreenWidth(currentMonitor);
+        int scaledMonitorH = UIUtil.getScaledScreenHeight(currentMonitor);
+        int w = Math.min(getSize().width, scaledMonitorW);
+        int h = Math.min(getSize().height, scaledMonitorH);
+        setSize(new Dimension(w, h));
+    }
+
+
+    //<editor-fold desc="MenuBarOwner interface implementation">
+    @Override
+    public JFrame getFrame() {
+        return this;
+    }
+
+    @Override
+    public Entity getEntity() {
+        return currentEditor().getEntity();
+    }
+
+    @Override
+    public String getFileName() {
+        return currentEditor().getFileName();
+    }
+
+    @Override
+    public boolean hasEntityNameChanged() {
+        return currentEditor().hasEntityNameChanged();
+    }
+
+    @Override
+    public void refreshMenuBar() {
+        menuBar.refreshMenuBar();
+    }
+
+    @Override
+    public MenuBar getMMLMenuBar() {
+        return menuBar;
+    }
+    //</editor-fold>
```

Comment:
I don't think IDE-specific formatting comments should be included in the main repo.